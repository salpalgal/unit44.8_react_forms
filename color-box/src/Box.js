import React, {userState} from "react"






const Box = ({width=10, height=10,color="blue",id,handleRemove})=>{
  
    const remove = ()=>handleRemove(id)
    return (
        <div>
            <div 
                style={{
                width: `${width}em`,
                height: `${height}em`,
                backgroundColor : color, 
            }}/>
            
            <button onClick={remove}>X</button>
        </div>
    )
}


export default Box;