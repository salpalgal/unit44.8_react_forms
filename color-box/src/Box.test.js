import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Box from "./Box";

it("works without crashing", function(){
  render(<Box />)
})

it("matches snapshot", function() {
    const { asFragment } = render(<Box />);
    expect(asFragment()).toMatchSnapshot();
  });