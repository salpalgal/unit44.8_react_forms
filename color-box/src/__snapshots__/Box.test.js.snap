// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      style="width: 10em; height: 10em; background-color: blue;"
    />
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
